src/arch/xtensa/include/xtensa/config/core.h

```diff
@@ -54,6 +54,14 @@
 
 #endif
 
+#if defined(CONFIG_ACP_7_0)
+
+#ifndef UINT32_C
+#define UINT32_C(x) x
```

Comment:
Please add a comment explaining why this is needed.

Do you really need BOTH `#ifdef ACP70` AND `#ifndef UINT32_C`?

Reply:
Yes, Actually ACP_7_0 platform uses the new 2023 xt-clang toolchain for XCC build. That need the defination of macro UINT32_C.

Reply:
Just realized there is already something similar a few lines above. So we have some toolchain-specific ugliness with four `#ifdef`, none of which actually checks any toolchain stuff... This looks verbose, messy and fragile. @paulstelian97 , @andyross, @dcpleung  can you help maybe?

Reply:
> Just realized there is already something similar a few lines above. So we have some toolchain-specific ugliness with four `#ifdef`, none of which actually checks any toolchain stuff... This looks duplicated, messy and fragile. @paulstelian97 , @andyross, @dcpleung can you help maybe?

gentle reminder

Reply:
IIRC, the old gcc-based xcc needs this. But newer xt-clang should be fine. Are you including `stdint.h` in correct place?

Reply:
Thanks @dcpleung , great point: both MTL and LNL have been using `xt-clang` for a long time. Neither needs this. So why should ACP7.0 need it?

Reply:
Unless we can take a look at their toolchain, it's hard to find out why. The only thing I can think of is that `stdint.h` is not being included.

Reply:
While Xtensa toolchains are designed for customization, they're not design for _that_ sort of "customization"! I mentioned MTL and LNL but IMX does not need this either.

Reply:
> Thanks @dcpleung , great point: both MTL and LNL have been using `xt-clang` for a long time. Neither needs this. So why should ACP7.0 need it?

But file tie.h (where the UINT32_C() is present at line 37-40 in src/platform/amd/acp_7_0/include/arch/xtensa/config/tie.h) is platform specific core file provided by xtensa. Can also observe that all tie.h files present in platforms MTL and LNL don't have the usage of the macro UINT32_C().

Reply:
> Unless we can take a look at their toolchain, it's hard to find out why. The only thing I can think of is that `stdint.h` is not being included.

Including the header file `stdint.h` in the `core.h` is raising an error as below.  Actually expecting the definition `#define UINT32_C(x) x`. But got `#define UINT32_C(x) x ##U` from the header file `stdint.h`.
 
```
/home/sof/src/arch/xtensa/include/xtensa/xtruntime-frames.h:128: Error: found 'U', expected: ')'
/home/sof/src/arch/xtensa/include/xtensa/xtruntime-frames.h:128: Error: found 'U', expected: ')'
/home/sof/src/arch/xtensa/include/xtensa/xtruntime-frames.h:128: Error: found 'U', expected: ')'
/home/sof/src/arch/xtensa/include/xtensa/xtruntime-frames.h:128: Error: found 'U', expected: ')'
/home/sof/src/arch/xtensa/include/xtensa/xtruntime-frames.h:128: Error: found 'U', expected: ')'
/home/sof/src/arch/xtensa/include/xtensa/xtruntime-frames.h:128: Error: junk at end of line, first unrecognized character is `U'
```

Reply:
Instead of defining `UINT32_C(x)`, maybe include `xtensa-types.h` instead? It is defined there on the toolchains I have.

Reply:
have tried including `xtensa-types.h` in file [core.h](https://github.com/thesofproject/sof/blob/main/src/arch/xtensa/include/xtensa/config/core.h)  at line 59.

Reply:
You can always do:
```
#if defined(__XT_CLANG__)
#include <xtensa/xtensa-types.h>
#endif
```
to limit it to xt-clang.

Reply:
If `defined(__XT_CLANG__)` works, then how about:

```C
#if defined(__XT_CLANG__)
#include <xtensa/xtensa-types.h>
#elif defined(__ZEPHYR__)
 ... code already above ...
#endif
```

Also: please submit this change in a separate PR because it affects all platforms. So it should not be buried and hidden in this giant, platform-specific PR.

Reply:
... or maybe just drop the `__ZEPHYR__` guard. It's not clear why @paulstelian97 added it in #7538.

In any case this is a complex toolchain compatibility issue that definitely deserves a separate PR.

Reply:
> You can always do:
> 
> ```
> #if defined(__XT_CLANG__)
> #include <xtensa/xtensa-types.h>
> #endif
> ```
> 
> to limit it to xt-clang.

Condition `#if defined(__XT_CLANG__)` don't work( i.e., goes False) for GCC build. Actually both XCC and GCC builds require the inclusion of `xtensa-types.h` .

Reply:
> ... or maybe just drop the `__ZEPHYR__` guard. It's not clear why @paulstelian97 added it in #7538.
> 
> In any case this is a complex toolchain compatibility issue that definitely deserves a separate PR.

Created a separate [PR](https://github.com/thesofproject/sof/pull/9413) for inclusion of xtensa-types.h file which has the macro UINT32_C.

Reply:
GCC would not have `__XT_CLANG__` declared so that should not affect the build... no?

I think we need clarification here. There are XCC (based on old GCC) and XT-CLANG (based on LLVM) compilers, and there is GCC which is not provided by Cadence. XCC build would not have `__XT_CLANG__` declared either. Or are you saying that older XT-CLANG toolchain are having problems with including `xtensa-types.h`?

Reply:
Thanks @dcpleung , can you please help transfer that question to newer, much smaller and focused #9413? It's hard enough to get attention to tricky issues like this, but it's much harder buried deep down this big, platform-specific PR named "ACP 7.0"